packages/hoppscotch-app/newstore/SSESession.ts

```diff
@@ -0,0 +1,185 @@
+import { pluck, distinctUntilChanged } from "rxjs/operators"
+import DispatchingStore, { defineDispatchers } from "./DispatchingStore"
+
+type SSERequest = {
+  endpoint: string
+  eventType: string
+}
+
+type SSESession = {
+  request: SSERequest
+  connectingState: boolean
+  connectionState: boolean
+  log: Array
+  socket
+}
+
+const defaultSSERequest: SSERequest = {
+  endpoint: "https://express-eventsource.herokuapp.com/events",
+  eventType: "data",
+}
+
+const defaultSSESession: SSESession = {
+  request: defaultSSERequest,
+  connectionState: false,
+  connectingState: false,
+  socket: null,
+  log: [],
+}
+
+const dispatchers = defineDispatchers({
+  setRequest(_, { newRequest }: { newRequest: SSERequest }) {
+    return {
+      request: newRequest,
+    }
+  },
+  setEndpoint(curr: SSESession, { newEndpoint }: { newEndpoint: string }) {
+    return {
+      request: {
+        eventType: curr.request.eventType,
+        endpoint: newEndpoint,
+      },
+    }
+  },
+  setEventType(curr: SSESession, { newType }: { newType: string }) {
+    return {
+      request: {
+        endpoint: curr.request.endpoint,
+        eventType: newType,
+      },
+    }
+  },
+  setSocket(_, { socket }) {
+    return {
+      socket,
+    }
+  },
+  setConnectionState(_, { state }: { state: boolean }) {
+    return {
+      connectionState: state,
+    }
+  },
+  setConnectingState(_, { state }: { state: boolean }) {
+    return {
+      connectingState: state,
+    }
+  },
+  setLog(_, { log }) {
+    return {
+      log,
+    }
+  },
+  addLogLine(curr: SSESession, { line }) {
+    return {
+      log: [...curr.log, line],
+    }
+  },
+})
+
+const SSESessionStore = new DispatchingStore(defaultSSESession, dispatchers)
+
+export function setSSERequest(newRequest?: SSERequest) {
+  SSESessionStore.dispatch({
+    dispatcher: "setRequest",
+    payload: {
+      newRequest: newRequest ?? defaultSSERequest,
+    },
+  })
+}
+
+export function setSSEEndpoint(newEndpoint: string) {
+  SSESessionStore.dispatch({
+    dispatcher: "setEndpoint",
+    payload: {
+      newEndpoint,
+    },
+  })
+}
+
+export function setSSEEventType(newType: string) {
+  SSESessionStore.dispatch({
+    dispatcher: "setEventType",
+    payload: {
+      newType,
+    },
+  })
+}
+
+export function setSSESocket(socket) {
```

Comment:
No type definition

Reply:
Also, I am not sure if the socket should be preserved in the store due to it not being suitable for persistence